### PR TITLE
Hamcore.c: handle memory allocation error

### PR DESCRIPTION
**Pull request by chipitsine**

### CODE REVIEW

**Review by davidebeatrici — APPROVED** [review]

File: Hamcore.c [hunk 1]
```diff
@@ -277,9 +281,21 @@ bool HamcoreBuild(const char *dst_path, const char *base_path, const char **src_
 
 		const size_t path_size = strlen(relative_path) + 1;
 		file->Path = malloc(path_size);
+		if (!file->Path)
+		{
+			free(buffer);
+			return false;
+		}
```
Comment (davidebeatrici) [inline Hamcore.c]:
```suggestion
		if (!file->Path)
		{
			free(compressed_files);
			free(buffer);
			return false;
		}
```

File: Hamcore.c [hunk 2]
```diff
@@ -277,9 +281,21 @@ bool HamcoreBuild(const char *dst_path, const char *base_path, const char **src_
...
+
 		memcpy(file->Path, relative_path, path_size);
 
 		compressed_file->Data = malloc(file->Size);
+		if (!compressed_file->Data)
+		{
+			free(buffer);
+			return false;
+		}
```
Comment (davidebeatrici) [inline Hamcore.c]:
```suggestion
		if (!compressed_file->Data)
		{
			free(compressed_files);
			free(buffer);
			return false;
		}
```

**Comment by davidebeatrici** [comment]
Could you squash the commits, please?

**Comment by chipitsine** [comment]
sure, will do soon

**Comment by chipitsine** [comment]
squashed

**Review by davidebeatrici — APPROVED** [review]

**Pull request merged**